### PR TITLE
os/board/rtl8720e: Add BT example for amebalite

### PR DESCRIPTION
**Pull request by Dracaryser**
add BT example for amebalite and modify related makefiles

### CODE REVIEW

**Comment by sunghan-chang** [comment]
@an4967 @taehwan-hwang Could you review this PR?

File: os/board/rtl8720e/src/component/bluetooth/api/Make.defs [hunk 1]
```diff
@@ -0,0 +1,55 @@
+###########################################################################
+#
+# Copyright 2020 Samsung Electronics All Rights Reserved.
```
Comment (sunghan-chang) [inline os/board/rtl8720e/src/component/bluetooth/api/Make.defs:3]: Could you update the year of copyright for Samsung License? 2020 -> 2022

Comment (Dracaryser) [reply os/board/rtl8720e/src/component/bluetooth/api/Make.defs:3]: Dear Mr. Chang, Sorry, I'll fix this.

File: os/board/rtl8720e/src/component/bluetooth/driver/hci/hci_if_rtk.c [hunk 2]
```diff
@@ -0,0 +1,372 @@
...
+	(void)context;
+	uint8_t task_run = 1, msg;
+	struct tx_queue_t *tx_queue = &hci_if_rtk.tx_queue;
+
+#if defined(CONFIG_BUILD_NONSECURE)
+	osif_create_secure_context(128);
+#endif
+
+	while (task_run) {
+		osif_msg_recv(hci_if_rtk.msg_q, &msg, 0xFFFFFFFF);
```
[the remark this answers is not in the document]
Comment (Dracaryser) [reply os/board/rtl8720e/src/component/bluetooth/driver/hci/hci_if_rtk.c:119]: Hi, I will fix these issues you mentioned above, thanks.

**Review by an4967 — CHANGES_REQUESTED** [review]
There are some error while building tc/audio config.
Please check them out.

File: os/board/rtl8720e/src/component/bluetooth/driver/hci/hci_if_rtk.c [hunk 3]
```diff
@@ -0,0 +1,390 @@
...
+	return true;
+}
+
+bool hci_if_confirm(uint8_t *buf)
+{
+#if 0//def CONFIG_AYNSC_HCI_INTF
+	/* TODO */
+#else
+	osif_mem_aligned_free(buf - (RESERVED_LEN - H4_HDR_LEN));
+#endif
```
Comment (taehwan-hwang) [inline os/board/rtl8720e/src/component/bluetooth/driver/hci/hci_if_rtk.c:388]:
```suggestion
	osif_mem_aligned_free(buf - (RESERVED_LEN - H4_HDR_LEN));
```

File: os/board/rtl8720e/src/component/bluetooth/driver/hci/hci_if_rtk_s.c [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+#include "hci_if_rtk_s.h"
+#include "hci/hci_process.h"
+#include "hci/hci_transport.h"
+#include "hci_platform.h"
+#include "hci_dbg.h"
+
+static struct {
+	uint8_t* tx_buf;
+    uint8_t  status;
+} hci_if_rtk = {0, 0};
```
Comment (taehwan-hwang) [inline os/board/rtl8720e/src/component/bluetooth/driver/hci/hci_if_rtk_s.c:17]:
```suggestion
static struct {
	uint8_t* tx_buf;
	uint8_t  status;
} hci_if_rtk = {0, 0};
```

File: os/board/rtl8720e/src/component/bluetooth/driver/hci/hci_if_rtk_s.c [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+	hci_uart_set_irq(UART_RX, 1);
+	
+    hci_if_rtk.status = 1;
+    open_cb(true);
+    return;
+
+bailout:
+    hci_if_rtk.status = 0;
+    open_cb(false);
+}
```
Comment (taehwan-hwang) [inline os/board/rtl8720e/src/component/bluetooth/driver/hci/hci_if_rtk_s.c:68]:
```suggestion
void hci_tp_open(HCI_OPEN_CB open_cb, HCI_RECV_IND rx_ind)
{
	if (hci_if_rtk.status)
	{
		HCI_ERR("\n\rHci Driver Already Open!");
		goto bailout;
	}

	/* BT Board Init */
	if (HCI_FAIL == hci_platform_init())
		goto bailout;

	/* HCI Transport */
	if (HCI_FAIL == hci_transport_open())
		goto bailout;

	/* HCI Transport Bridge to StandAlone */
	hci_transport_set_get_buf(hci_sa_recv_get_buf);
	hci_transport_set_recv(hci_sa_recv);

	/* HCI UART Bridge to Transport */
	hci_uart_set_rx_ind(hci_transport_recv_ind);
	hci_uart_set_irq(UART_RX, 1);

	if (HCI_FAIL == hci_process())
		goto bailout;

	/* HCI UART Bridge to NULL */
	hci_uart_set_irq(UART_RX, 0);
	hci_uart_set_rx_ind(NULL);

	/* HCI Transport Close and free */
	if (HCI_FAIL == hci_transport_close() || 
		HCI_FAIL == hci_transport_free())
		goto bailout;

	/* HCI Transport Bridge to RTK Stack 
	* (Stop and Start rx_ind for this Moment)
	*/
	hci_uart_set_rx_ind(rx_ind);
	hci_uart_set_irq(UART_RX, 1);

	hci_if_rtk.status = 1;
	open_cb(true);
	return;

bailout:
    hci_if_rtk.status = 0;
    open_cb(false);
}
```

Comment (taehwan-hwang) [inline os/board/rtl8720e/src/component/bluetooth/driver/hci/hci_if_rtk_s.c:68]:
```suggestion
void hci_tp_open(HCI_OPEN_CB open_cb, HCI_RECV_IND rx_ind)
{
	if (hci_if_rtk.status)
	{
		HCI_ERR("\n\rHci Driver Already Open!");
		goto bailout;
	}

	/* BT Board Init */
	if (HCI_FAIL == hci_platform_init())
		goto bailout;

	/* HCI Transport */
	if (HCI_FAIL == hci_transport_open())
		goto bailout;

	/* HCI Transport Bridge to StandAlone */
	hci_transport_set_get_buf(hci_sa_recv_get_buf);
	hci_transport_set_recv(hci_sa_recv);

	/* HCI UART Bridge to Transport */
	hci_uart_set_rx_ind(hci_transport_recv_ind);
	hci_uart_set_irq(UART_RX, 1);

	if (HCI_FAIL == hci_process())
		goto bailout;

	/* HCI UART Bridge to NULL */
	hci_uart_set_irq(UART_RX, 0);
	hci_uart_set_rx_ind(NULL);

	/* HCI Transport Close and free */
	if (HCI_FAIL == hci_transport_close() || 
		HCI_FAIL == hci_transport_free())
		goto bailout;

	/* HCI Transport Bridge to RTK Stack 
	* (Stop and Start rx_ind for this Moment)
	*/
	hci_uart_set_rx_ind(rx_ind);
	hci_uart_set_irq(UART_RX, 1);

	hci_if_rtk.status = 1;
	open_cb(true);
	return;

bailout:
    hci_if_rtk.status = 0;
    open_cb(false);
}
```

Comment (taehwan-hwang) [inline os/board/rtl8720e/src/component/bluetooth/driver/hci/hci_if_rtk_s.c:68]:
```suggestion
void hci_tp_open(HCI_OPEN_CB open_cb, HCI_RECV_IND rx_ind)
{
	if (hci_if_rtk.status)
	{
		HCI_ERR("\n\rHci Driver Already Open!");
		goto bailout;
	}

	/* BT Board Init */
	if (HCI_FAIL == hci_platform_init())
		goto bailout;

	/* HCI Transport */
	if (HCI_FAIL == hci_transport_open())
		goto bailout;

	/* HCI Transport Bridge to StandAlone */
	hci_transport_set_get_buf(hci_sa_recv_get_buf);
	hci_transport_set_recv(hci_sa_recv);

	/* HCI UART Bridge to Transport */
	hci_uart_set_rx_ind(hci_transport_recv_ind);
	hci_uart_set_irq(UART_RX, 1);

	if (HCI_FAIL == hci_process())
		goto bailout;

	/* HCI UART Bridge to NULL */
	hci_uart_set_irq(UART_RX, 0);
	hci_uart_set_rx_ind(NULL);

	/* HCI Transport Close and free */
	if (HCI_FAIL == hci_transport_close() || 
		HCI_FAIL == hci_transport_free())
		goto bailout;

	/* HCI Transport Bridge to RTK Stack 
	* (Stop and Start rx_ind for this Moment)
	*/
	hci_uart_set_rx_ind(rx_ind);
	hci_uart_set_irq(UART_RX, 1);

	hci_if_rtk.status = 1;
	open_cb(true);
	return;

bailout:
    hci_if_rtk.status = 0;
    open_cb(false);
}
```

Comment (taehwan-hwang) [inline os/board/rtl8720e/src/component/bluetooth/driver/hci/hci_if_rtk_s.c:68]:
```suggestion
void hci_tp_open(HCI_OPEN_CB open_cb, HCI_RECV_IND rx_ind)
{
	if (hci_if_rtk.status)
	{
		HCI_ERR("\n\rHci Driver Already Open!");
		goto bailout;
	}

	/* BT Board Init */
	if (HCI_FAIL == hci_platform_init())
		goto bailout;

	/* HCI Transport */
	if (HCI_FAIL == hci_transport_open())
		goto bailout;

	/* HCI Transport Bridge to StandAlone */
	hci_transport_set_get_buf(hci_sa_recv_get_buf);
	hci_transport_set_recv(hci_sa_recv);

	/* HCI UART Bridge to Transport */
	hci_uart_set_rx_ind(hci_transport_recv_ind);
	hci_uart_set_irq(UART_RX, 1);

	if (HCI_FAIL == hci_process())
		goto bailout;

	/* HCI UART Bridge to NULL */
	hci_uart_set_irq(UART_RX, 0);
	hci_uart_set_rx_ind(NULL);

	/* HCI Transport Close and free */
	if (HCI_FAIL == hci_transport_close() || 
		HCI_FAIL == hci_transport_free())
		goto bailout;

	/* HCI Transport Bridge to RTK Stack 
	* (Stop and Start rx_ind for this Moment)
	*/
	hci_uart_set_rx_ind(rx_ind);
	hci_uart_set_irq(UART_RX, 1);

	hci_if_rtk.status = 1;
	open_cb(true);
	return;

bailout:
    hci_if_rtk.status = 0;
    open_cb(false);
}
```

Comment (taehwan-hwang) [inline os/board/rtl8720e/src/component/bluetooth/driver/hci/hci_if_rtk_s.c:68]:
```suggestion
	/* HCI UART Bridge to NULL */
	hci_uart_set_irq(UART_RX, 0);
	hci_uart_set_rx_ind(NULL);

	/* HCI Transport Close and free */
	if (HCI_FAIL == hci_transport_close() || 
		HCI_FAIL == hci_transport_free())
		goto bailout;

	/* HCI Transport Bridge to RTK Stack 
	* (Stop and Start rx_ind for this Moment)
	*/
	hci_uart_set_rx_ind(rx_ind);
	hci_uart_set_irq(UART_RX, 1);

	hci_if_rtk.status = 1;
	open_cb(true);
	return;

bailout:
    hci_if_rtk.status = 0;
    open_cb(false);
}
```

File: os/board/rtl8720e/src/component/bluetooth/driver/hci/hci_if_rtk_s.c [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+		HCI_FAIL == hci_uart_free())
+		goto bailout;
+
+    hci_if_rtk.status = 0;
+    HCI_PRINT("\n\rHci Driver Closed!");
+	return;
+
+bailout:
+    hci_if_rtk.status = 1;
+    HCI_PRINT("\n\rHci Driver Close Failed!");
```
Comment (taehwan-hwang) [inline os/board/rtl8720e/src/component/bluetooth/driver/hci/hci_if_rtk_s.c:86]:
```suggestion
	if (!hci_if_rtk.status)
		goto bailout;

	/* Platform Deinit First and free */
	if (HCI_FAIL == hci_platform_deinit() || 
		HCI_FAIL == hci_uart_free())
		goto bailout;

	hci_if_rtk.status = 0;
	HCI_PRINT("\n\rHci Driver Closed!");
	return;

bailout:
	hci_if_rtk.status = 1;
	HCI_PRINT("\n\rHci Driver Close Failed!");
```

File: os/board/rtl8720e/src/component/bluetooth/driver/hci/hci_if_rtk_s.c [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+uint16_t hci_tp_recv(uint8_t *p_buf, uint16_t size)
+{
+    return hci_uart_read(p_buf, size);
+}
+
+void hci_tp_config(uint8_t *p_buf, uint16_t len)
+{
+    (void)p_buf;
+    (void)len;
+}
```
Comment (taehwan-hwang) [inline os/board/rtl8720e/src/component/bluetooth/driver/hci/hci_if_rtk_s.c:123]:
```suggestion
bool hci_tp_send(uint8_t *p_buf, uint16_t len, HCI_SEND_CB tx_cb)
{
	if (len != hci_uart_send(p_buf, len))
		return false;

	if (tx_cb)
		tx_cb();

	return true;
}

uint16_t hci_tp_recv(uint8_t *p_buf, uint16_t size)
{
	return hci_uart_read(p_buf, size);
}

void hci_tp_config(uint8_t *p_buf, uint16_t len)
{
	(void)p_buf;
	(void)len;
}
```

**Comment by Dracaryser** [comment]
Hi @an4967 , I didn't modify any tc/audio config, and other pull requests have same error while building tc/audio config in recent days. Could you please help me check it again? Thanks.

File: os/board/rtl8720e/src/component/bluetooth/driver/hci/hci_if_rtk_s.c [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+    hci_if_rtk.status = 0;
+    HCI_PRINT("\n\rHci Driver Closed!");
+	return;
+
+bailout:
+    hci_if_rtk.status = 1;
+    HCI_PRINT("\n\rHci Driver Close Failed!");
+}
+
+uint8_t flag_for_hci_trx = 0;
```
Comment (taehwan-hwang) [inline os/board/rtl8720e/src/component/bluetooth/driver/hci/hci_if_rtk_s.c:89]: How about use static keyword to make the variable local to single files.
```suggestion
static uint8_t flag_for_hci_trx = 0;
```

**Comment by an4967** [comment]
> Hi @an4967 , I didn't modify any tc/audio config, and other pull requests have same error while building tc/audio config in recent days. Could you please help me check it again? Thanks.

You can check the causes of building errors via below link.
I think they occurred because having multiple definitions of some variables.

https://app.circleci.com/pipelines/github/Samsung/TizenRT/1783/workflows/2a93f6a6-3b97-4f26-8769-fe34cf276cb5/jobs/14988

**Comment by Dracaryser** [comment]
> > Hi @an4967 , I didn't modify any tc/audio config, and other pull requests have same error while building tc/audio config in recent days. Could you please help me check it again? Thanks.
> 
> You can check the causes of building errors via below link. I think they occurred because having multiple definitions of some variables.
> 
> https://app.circleci.com/pipelines/github/Samsung/TizenRT/1783/workflows/2a93f6a6-3b97-4f26-8769-fe34cf276cb5/jobs/14988

I've checked the causes of error, and I think my commits have no relationship to it. I found that other one's pull requests have same error these days and their causes of error are same as mine. So may be it's a system issue?

File: os/board/rtl8720e/src/component/bluetooth/driver/hci/hci_if_rtk_s.c (hunk 8 above)
Comment (Dracaryser) [reply os/board/rtl8720e/src/component/bluetooth/driver/hci/hci_if_rtk_s.c:89]: Hi @taehwan-hwang , hci_if_rtk_s.c is actually not used, I will remove unused files.

**Comment by sunghan-chang** [comment]
@Dracaryser @an4967  Sorry for late reply. It's come from toolchain update and was resolved already in the master. Once PR #5558 , #5561 , #5562 are merged, we will sync this branch with the master. Then it will disappear. Please don't look at the CI build errors on audio and tc.

**Comment by taehwan-hwang** [comment]
Please resolve conflicts

**Comment by Dracaryser** [comment]
> Please resolve conflicts

Sorry, conflicts have been solved.

**Review by an4967 — APPROVED** [review]
LGTM

**Comment by Dracaryser** [comment]
> There are still some conflicts. Please check them.



> There are still some conflicts. Please check them.

Could you please show me where the conflicts are? There is no conflicts showed on the pull request page. Thanks.

**Comment by an4967** [comment]
@Dracaryser 
There is no comment about this in the System.
Can you follow below steps?

```
Step 1: From your project repository, check out a new branch and test the changes.

git checkout -b Dracaryser-Lite_Commit RTL8720E_Support
git pull https://github.com/Dracaryser/TizenRT.git Lite_Commit

Step 2: Merge the changes and update on GitHub.

git checkout RTL8720E_Support
git merge --no-ff Dracaryser-Lite_Commit
git push origin RTL8720E_Support
```

**Comment by Dracaryser** [comment]
> 

May I close this pull request and review my changes then raise PR again?

**Comment by sunghan-chang** [comment]
Because of two new commits, this PR can't be merged. System shows below
```
This branch cannot be rebased due to conflicts
```

[Merge branch 'Lite_Commit' into RTL8720E_Support](https://github.com/Samsung/TizenRT/pull/5558/commits/f097ef7093a98902c5f88938bc380d569e538ce5)
[f097ef7](https://github.com/Samsung/TizenRT/pull/5558/commits/f097ef7093a98902c5f88938bc380d569e538ce5)

[Merge pull request](https://github.com/Samsung/TizenRT/pull/5558/commits/45fa8020e28bea366c08a97bfdea02f7de9c2056) https://github.com/Dracaryser/TizenRT/pull/1 [from Dracaryser/RTL8720E_Support](https://github.com/Samsung/TizenRT/pull/5558/commits/45fa8020e28bea366c08a97bfdea02f7de9c2056) 
[45fa802](https://github.com/Samsung/TizenRT/pull/5558/commits/45fa8020e28bea366c08a97bfdea02f7de9c2056)

You need to rebase your commits.

**Pull request closed without merging**